### ISSUE TITLE
bug: `hide-get-premium` patch fails when not including other patches

### ISSUE THREAD

**Issue by hoang-himself** [inotia00/ReVanced_Extended#386]
### Type

Error while patching

### Application

YouTube Music v5.44.54, v5.45.52, v5.45.53, v5.46.52

### Bug description

While patching

### Steps to reproduce

Step 1: include `hide-get-premium` and patch

### Relevant log output

```shell
Initializing installer
Creating working directory
Copying original apk
Unpacking input apk
Reading dex files
Decoding AndroidManifest.xml only, because resources are not needed
Merging integrations
An error occurred! Aborting
Error:
java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Constructor.newInstance0(Native Method)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:318)
	at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:37)
	at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:355)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:129)
	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda$20(MainActivity.kt:327)
	at app.revanced.manager.flutter.MainActivity.$r8$lambda$VzOqgSVOualTn6WBlAKw3muxHF8(Unknown Source:0)
	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda13.run(Unknown Source:24)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: kotlin.UninitializedPropertyAccessException: lateinit property resourceMappings has not been initialized
	at app.revanced.patches.shared.patch.mapping.ResourceMappingPatch$Companion.getResourceMappings$revanced_patches(ResourceMappingPatch.kt:20)
	at app.revanced.patches.music.layout.premium.patch.HideGetPremiumPatch.<init>(HideGetPremiumPatch.kt:40)
	... 11 more
```

### Device Environment

Samsung Galaxy A71, OneUI 5.0, Android 13, arm64-v8a

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by ILoveOpenSourceApplications** [comment]
Are you a rooted user? If not please include the Music-MicroG-Support and try repatching again.

**Comment by hoang-himself** [comment]
I am not rooted. I tried applying both `Music MicroG Support` and `Hide Get Premium` and I still encounter the bug, same log output.

**Comment by inotia00** [comment]
ah, I missed this just before the final build
This will be fixed in the next release

**Comment by ILoveOpenSourceApplications** [comment]
> ah, I missed this just before the final build This will be fixed in the next release

Yes, it seems hide-get-premium requires some other patches to be included with it or else it keeps failing.

**Comment by DrimTim2020** [comment]
> ### Type
> Error while patching
> 
> ### Application
> YouTube Music v5.44.54, v5.45.52, v5.45.53, v5.46.52
> 
> ### Bug description
> While patching
> 
> ### Steps to reproduce
> Step 1: include `hide-get-premium` and patch
> 
> ### Relevant log output
> ```shell
> Initializing installer
> Creating working directory
> Copying original apk
> Unpacking input apk
> Reading dex files
> Decoding AndroidManifest.xml only, because resources are not needed
> Merging integrations
> An error occurred! Aborting
> Error:
> java.lang.reflect.InvocationTargetException
> 	at java.lang.reflect.Constructor.newInstance0(Native Method)
> 	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
> 	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:318)
> 	at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:37)
> 	at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:355)
> 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
> 	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:129)
> 	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda$20(MainActivity.kt:327)
> 	at app.revanced.manager.flutter.MainActivity.$r8$lambda$VzOqgSVOualTn6WBlAKw3muxHF8(Unknown Source:0)
> 	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda13.run(Unknown Source:24)
> 	at java.lang.Thread.run(Thread.java:1012)
> Caused by: kotlin.UninitializedPropertyAccessException: lateinit property resourceMappings has not been initialized
> 	at app.revanced.patches.shared.patch.mapping.ResourceMappingPatch$Companion.getResourceMappings$revanced_patches(ResourceMappingPatch.kt:20)
> 	at app.revanced.patches.music.layout.premium.patch.HideGetPremiumPatch.<init>(HideGetPremiumPatch.kt:40)
> 	... 11 more
> ```
> 
> ### Screenshots or videos
> _No response_
> 
> ### Solution
> _No response_
> 
> ### Additional context
> _No response_
> 
> ### Device Environment
> Samsung Galaxy A71, OneUI 5.0, Android 13, arm64-v8a
> 
> ### Acknowledgements
> * [x]  I have searched the existing issues and this is a new and no duplicate or related to another open issue.
> * [x]  I have written a short but informative title.
> * [x]  I filled out all of the requested information in this issue properly.
the patch does not work for a long time, I reported this, but the author does not care about it.
Be patient with it as it is.

**Comment by DrimTim2020** [comment]
> ### Тип
> Ошибка при прошивке
> 
> ### Приложение
> YouTube Музыка v5.44.54, v5.45.52, v5.45.53, v5.46.52
> 
> ### Описание ошибки
> При исправлении
> 
> ### Действия по воспроизведению
> Шаг 1: включить `hide-get-premium`и исправить
> 
> ### Соответствующий вывод журнала
> ```shell
> Initializing installer
> Creating working directory
> Copying original apk
> Unpacking input apk
> Reading dex files
> Decoding AndroidManifest.xml only, because resources are not needed
> Merging integrations
> An error occurred! Aborting
> Error:
> java.lang.reflect.InvocationTargetException
> 	at java.lang.reflect.Constructor.newInstance0(Native Method)
> 	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
> 	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:318)
> 	at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:37)
> 	at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:355)
> 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
> 	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:129)
> 	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda$20(MainActivity.kt:327)
> 	at app.revanced.manager.flutter.MainActivity.$r8$lambda$VzOqgSVOualTn6WBlAKw3muxHF8(Unknown Source:0)
> 	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda13.run(Unknown Source:24)
> 	at java.lang.Thread.run(Thread.java:1012)
> Caused by: kotlin.UninitializedPropertyAccessException: lateinit property resourceMappings has not been initialized
> 	at app.revanced.patches.shared.patch.mapping.ResourceMappingPatch$Companion.getResourceMappings$revanced_patches(ResourceMappingPatch.kt:20)
> 	at app.revanced.patches.music.layout.premium.patch.HideGetPremiumPatch.<init>(HideGetPremiumPatch.kt:40)
> 	... 11 more
> ```
> 
> ### Скриншоты или видео
> _Нет ответа_
> 
> ### Решение
> _Нет ответа_
> 
> ### Дополнительный контекст
> _Нет ответа_
> 
> ### Среда устройства
> Samsung Galaxy A71, OneUI 5.0, Android 13, arm64-v8a
> 
> ### Благодарности
> * [x] Я просмотрел существующие проблемы, и это новая проблема, не дублирующаяся и не связанная с другой открытой проблемой.
> * [x] Я написал короткое, но информативное название.
> * [x] Я правильно заполнил всю запрашиваемую информацию в этом вопросе.

The patch does not work, I wrote about this to the author for a long time, but the author ignores this and simply closes the topic.
Well, if he is a lol or a clown, then this is so in his spirit, he can do so.

**Comment by ILoveOpenSourceApplications** [comment]
> The patch does not work, I wrote about this to the author for a long time, but the author ignores this and simply closes the topic. Well, if he is a lol or a clown, then this is so in his spirit, he can do so.

First of all, I don't see any issues open or closed regarding this topic specifically. Secondly, why are you so aggressive towards the developer when he's giving his all to fix whatever issues that's being reported here pushing more patch updates than the official ReVanced, which isn't bad itself as they're assuring more stable patch releases, but hey he's a solo developer and is working despite announcing that's he's going on a vacation. Be grateful.

**Comment by inotia00** [comment]
@DrimTim2020 You don't seem to know what this patch means

The `hide-get-premium` patch is a patch that removes this banner.
![music](https://user-images.githubusercontent.com/108592928/222528814-4558e521-6e41-462c-babc-085370bcd7e0.png)

And, when all patches are included, this patch works just fine.

**Comment by DrimTim2020** [comment]
> > The patch does not work, I wrote about this to the author for a long time, but the author ignores this and simply closes the topic. Well, if he is a lol or a clown, then this is so in his spirit, he can do so.
> 
> First of all, I don't see any issues open or closed regarding this topic specifically. Secondly, why are you so aggressive towards the developer when he's giving his all to fix whatever issues that's being reported here pushing more patch updates than the official ReVanced, which isn't bad itself as they're assuring more stable patch releases, but hey he's a solo developer and is working despite announcing that's he's going on a vacation. Be grateful.

You're not lost in time, are you?
I think it's you all Indians who react very aggressively !!!!!!!
After my post, when I reported that it simply did not work, he took it and simply closed the topic, then how should I react.
Can you stop being stupid and write something smarter?
It's just that this is not going nowhere.

**Comment by inotia00** [comment]
This is a place to report issues, not a place for discussion.
please stop the dispute

Also, please refrain from actions that may cause a dispute.

**Comment by inotia00** [comment]
fixed in revanced-patches-v2.164.15
